orm/hibernate-orm-4/src/test/java/org/hibernate/bugs/ORMStandaloneJpaUnitTestCase.java

```diff
@@ -0,0 +1,203 @@
+			this.name = name;
+		}
+
+		@Override
+		public URL getPersistenceUnitRootUrl() {
+			return null;
+		}
+
+		@Override
+		public String getName() {
+			return name;
+		}
+
+		@Override
+		public String getProviderClassName() {
+			return HibernatePersistenceProvider.class.getName();
+		}
+
+		@Override
+		public boolean isUseQuotedIdentifiers() {
+			return false;
+		}
+
+		@Override
+		public boolean isExcludeUnlistedClasses() {
+			return false;
+		}
+
+		@Override
+		public PersistenceUnitTransactionType getTransactionType() {
+			return null;
+		}
+
+		@Override
+		public ValidationMode getValidationMode() {
+			return null;
+		}
+
+		@Override
+		public SharedCacheMode getSharedCacheMode() {
+			return null;
+		}
+
+		@Override
+		public List<String> getManagedClassNames() {
+			return null;
+		}
+
+		@Override
+		public List<String> getMappingFileNames() {
+			return null;
+		}
+
+		@Override
+		public List<URL> getJarFileUrls() {
+			return null;
+		}
+
+		@Override
+		public Object getNonJtaDataSource() {
+			return null;
+		}
+
+		@Override
+		public Object getJtaDataSource() {
+			return null;
+		}
+
+		@Override
+		public Properties getProperties() {
+			return null;
+		}
+
+		@Override
+		public ClassLoader getClassLoader() {
+			return null;
+		}
+
+		@Override
+		public void pushClassTransformer(List<String> entityClassNames) {
+
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	private Map buildSettings() {
+		Map settings = getConfig();
+		String[] mappings = getMappings();
+		if ( mappings != null ) {
+			settings.put( AvailableSettings.HBXML_FILES, StringHelper.join( ",", mappings ) );
+		}
+
+		settings.put( org.hibernate.cfg.AvailableSettings.HBM2DDL_AUTO, "update" );
+		settings.put(org.hibernate.cfg.AvailableSettings.SHOW_SQL, "true");
+		settings.put(org.hibernate.cfg.AvailableSettings.FORMAT_SQL, "true");
+
+		return settings;
+	}
+
+	private Map getConfig() {
+		Map<Object, Object> config = Environment.getProperties();
+		ArrayList<Class> classes = new ArrayList<Class>();
+
+		classes.addAll( Arrays.asList( getAnnotatedClasses() ) );
+		config.put( AvailableSettings.LOADED_CLASSES, classes );
+		for ( Map.Entry<Class, String> entry : getCachedClasses().entrySet() ) {
+			config.put( AvailableSettings.CLASS_CACHE_PREFIX + "." + entry.getKey().getName(), entry.getValue() );
+		}
+		for ( Map.Entry<String, String> entry : getCachedCollections().entrySet() ) {
+			config.put( AvailableSettings.COLLECTION_CACHE_PREFIX + "." + entry.getKey(), entry.getValue() );
+		}
+		if ( getEjb3DD().length > 0 ) {
+			ArrayList<String> dds = new ArrayList<String>();
+			dds.addAll( Arrays.asList( getEjb3DD() ) );
+			config.put( AvailableSettings.XML_FILE_NAMES, dds );
+		}
+		return config;
+	}
+
+	private Map<Class, String> getCachedClasses() {
+		return new HashMap<Class, String>();
+	}
+
+    private Map<String, String> getCachedCollections() {
+		return new HashMap<String, String>();
+	}
+
+    private String[] getEjb3DD() {
```

Comment:
It sets `AvailableSettings.XML_FILE_NAMES` configuration property and I took it from the hibernate-entitymanager and I wanted to make it as close as possible to those tests in case we want to migrate tests to that repository as well.